Performance measurement (#19)
eclipsesource/theia: scripts/performance/README.md

# Performance measurements

> This directory contains a script that measures the performance of theia.

Reviewer: ```suggestion
This directory contains a script that measures the performance of Theia.
```

> Currently the support is limited to measuring the `browser-app`s startup time using the `Largest contentful Paint (LCP)` value.

Reviewer: ```suggestion
Currently the support is limited to measuring the `browser-app`s startup time using the `Largest contentful paint (LCP)` value.
```

## Running the script

### Prerequisites

> To run the script the theia backend needs to be started.

Reviewer: ```suggestion
To run the script the Theia backend needs to be started.
```

This can either be done with the `Launch Browser Backend` launch config or by running `yarn start` in the `examples/browser-app` directory.

> Additionally, the root package.json file contains node scripts that startup the backend as well as the script. Like the `performance:startup` script.

Reviewer: ```suggestion
Additionally, the root package.json file contains node scripts that startup the backend as well as the script.
See the `performance:startup` script.
```

### Executing the script

The script can be exectued using `node measure-performance.js` in this directory.

The script accepts the following parameters:

-   `--name`: Specify a name for the current measurement (default: `Measurement`)

> -   `--url`: Point theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<absolutePathToWorkspaceFolder>`)

Reviewer: ```suggestion
-   `--url`: Point Theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<absolutePathToWorkspaceFolder>`)
```

# Performance measurements

This directory contains a script that measures the performance of Theia.
Currently the support is limited to measuring the `browser-app`s startup time using the `Largest contentful paint (LCP)` value.

## Running the script

> ### Prerequisites
> 
> To run the script the Theia backend needs to be started.
> This can either be done with the `Launch Browser Backend` launch config or by running `yarn start` in the `examples/browser-app` directory.
> 
> Additionally, the root package.json file contains node scripts that startup the backend as well as the script.
> See the `performance:startup` script.

Reviewer: ```suggestion
### Quick Start

Execute `yarn run performance:startup` in the root directory.

### Prerequisites

To run the performance tests the Theia backend needs to be started.
This can either be done with the `Launch Browser Backend` launch config or by running `yarn start` in the `examples/browser-app` directory.
```

### Executing the script

The script can be exectued using `node measure-performance.js` in this directory.

> The script accepts the following parameters:

Reviewer: ```suggestion
The script accepts the following optional parameters:
```

-   `--name`: Specify a name for the current measurement (default: `Measurement`)

> -   `--url`: Point Theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<absolutePathToWorkspaceFolder>`)

Reviewer: Default is workspace next to the script, right?
```suggestion
-   `--url`: Point Theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<pathToMeasurementScript>/workspace`)
```

# Performance measurements

This directory contains a script that measures the performance of Theia.
Currently the support is limited to measuring the `browser-app`'s startup time using the `Largest contentful paint (LCP)` value.

## Running the script

### Quick Start

Execute `yarn run performance:startup` in the root directory to startup the backend and execute the script.

### Prerequisites

To run the script the Theia backend needs to be started.
This can either be done with the `Launch Browser Backend` launch config or by running `yarn start` in the `examples/browser-app` directory.

### Executing the script

The script can be exectued using `node measure-performance.js` in this directory.

The script accepts the following optional parameters:

-   `--name`: Specify a name for the current measurement (default: `Measurement`)

> -   `--url`: Point Theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<pathToMeasurementScript>/workspace>`)

Reviewer: ```suggestion
-   `--url`: Point Theia to a url for example for specifying a specifc workspace (default: `http://localhost:3000/#/<pathToMeasurementScript>/workspace`)
```

> -   `--folder`: Folder name for the generated tracing files in the `profiles` folder (default: `profiles/profile`)

Reviewer: The actual value of the parameter is `profile`
```suggestion
-   `--folder`: Folder name for the generated tracing files in the `profiles` folder (default: `profile`)
```
